scripts/gis-material-update/validate-deploy/modules/common.py

```diff
@@ -0,0 +1,91 @@
+import math
+from sqlalchemy import create_engine, text
+    
+def deploy(pg_conn_uri, tormays_table_org, tormays_table_temp, logger=None):
+    # validate data amount: is it between given limits
+    delete_result = None
+    engine = create_engine(pg_conn_uri)
+    with engine.connect() as connection:
+        transaction = connection.begin()
+        delete_sql = "delete from " + tormays_table_org
+        try:
+            logger.info("Deploy ...")
+            delete_result = connection.execute(text(delete_sql))
+            logger.info(tormays_table_org + ": " + f"{delete_result.rowcount} rows deleted.")
+        except Exception as e:
+            # Transaction implicitly rolls back if an exception occurred within the "with" block
+            logger.error(f"Error: {e}")
+        
+        if delete_result:    
+            copy_sql = "insert into " + tormays_table_org + " select * from " + tormays_table_temp
+            try:
+                copy_result = connection.execute(text(copy_sql))
+                transaction.commit()
+                logger.info(tormays_table_org + ": " + f"{copy_result.rowcount} rows inserted.")
+            except Exception as e:
+                # Transaction implicitly rolls back if an exception occurred within the "with" block
+                logger.error(f"Error: {e}")
+            
+def validate_data_count_limits(module, pg_conn_uri, tormays_table_org, tormays_table_temp, validate_limit_min, validate_limit_max, logger=None):
+    # validate data amount: is it between given limits
+    tormays_result = None
+    temp_result = None
+    retval = None
+    old_amount = None
+    new_amount = None
+    tormays_sql = "select count(*) lkm from " + tormays_table_org + " where geometry is not null"
+    temp_sql = "select count(*) lkm from " + tormays_table_temp + " where geometry is not null"
+    engine = create_engine(pg_conn_uri)
+    with engine.connect() as connection:
+        try:
+            tormays_result = connection.execute(text(tormays_sql)).fetchall()
+        except Exception as e:
+            # Transaction implicitly rolls back if an exception occurred within the "with" block
```

Comment:
There's no transaction here.

Reply:
There is only "select" -clauses used here so no need for transaction.

Reply:
I know, but the comment is misleading.

Reply:
Comment removed.